content/inference/pytorch.md

```diff
@@ -0,0 +1,410 @@
+
+#### Example Use Cases 
+The ZZ\\(\rightarrow\\)4b analysis utilizes trained PyTorch models via ONNX in CMSSW (see the corresponding [repo](https://github.com/patrickbryant/ZZ4b/blob/master/README.md)). Briefly, they run ONNX in CMSSW_11_X via the CMSSW package `PhysicsTools/ONNXRuntime`, using it to define a [multiClassifierONNX](https://github.com/patrickbryant/ZZ4b/blob/5931a21d8005683e23166c0b44b9594b52ad1126/nTupleAnalysis/interface/multiClassifierONNX.h) class. This multiclassifier is capable of loading pre-trained PyTorch models specified by a `modelFile` string as follows:
+
+``` C++
+#include "PhysicsTools/ONNXRuntime/interface/ONNXRuntime.h"
+
+std::unique_ptr<cms::Ort::ONNXRuntime> model;
+Ort::SessionOptions* session_options = new Ort::SessionOptions();
+session_options->SetIntraOpNumThreads(1);
+model = std::make_unique<cms::Ort::ONNXRuntime>(modelFile, session_options);
+```
+
+### Via Triton
+Coprocessors (GPUs, FPGAs, etc.) are frequently used to accelerate ML operations such as inference and training. In the 'as-a-service' paradigm, users can access cloud-based applications through lightweight client inferfaces. The Services for Optimized Network Inference on Coprocessors ([SONIC](https://github.com/cms-sw/cmssw/tree/master/HeterogeneousCore/SonicCore)) framework implements this paradigm in CMSSW, allowing the optimal integration of GPUs into event processing workflows. One powerful implementation of SONIC is the the NVIDIA Triton Inference Server, which is flexible with respect to ML framework, storage source, and hardware infrastructure. For more details, see the corresponding [NVIDIA developer blog entry](https://developer.nvidia.com/blog/scaling-inference-in-high-energy-particle-physics-at-fermilab-using-nvidia-triton-inference-server/). 
+
+A Graph Attention Network (GAN) is available via Triton in CMSSW, and can be accessed here: https://github.com/cms-sw/cmssw/tree/master/HeterogeneousCore/SonicTriton/test
+
+## Training Tips
+- When instantiating a `DataLoader`, `shuffle=True` should be enabled for training data but not for validation and testing data. At each training epoch, this will vary the order of data objects in each batch; accordingly, it is not efficient to load the full dataset (in its original ordering) into GPU  memory before training. Instead, enable `num_workers>1`; this allows the `DataLoader` to load batches to the GPU as they're prepared. Note that this launches muliple processerson the CPU. For more information, see a corresponding [discussion](https://discuss.pytorch.org/t/keras-trains-significantly-faster-than-pytorch-for-simple-network/124303/5) in the PyTorch forum. 
```

Comment:
"multiple processerson the" -> "multiple processors on"

Do you mean processors or threads?